cmd/juju/application/deploy.go

```diff
@@ -1169,6 +1170,24 @@ func (c *DeployCommand) validateCharmSeries(series string) error {
 	return model.ValidateSeries(modelType, series)
 }
 
+func (c *DeployCommand) validateResourcesNeededForLocalDeploy(charmMeta *charm.Meta) error {
+	modelType, err := c.ModelType()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if modelType != model.CAAS {
+		return nil
+	}
+	for resName, resMeta := range charmMeta.Resources {
+		if resMeta.Type == resource.TypeContainerImage {
+			if _, ok := c.Resources[resName]; !ok {
+				return errors.NotValidf("must specify OCI images for local charm (%s)", resName)
```

Comment:
error message implies the charm name is <resname>
also, please loop over all resources and print a list of all that are missing in one go

could try
"local charm missing OCI images for %v", strings.Join(missing, ", ")